admin/services/check-start.service.js

```diff
@@ -61,24 +67,83 @@ checkStartService.prepareCheck = async function (pupilIds, dfeNumber, schoolId,
   const checks = []
   for (let pid of pupilIds) {
     const usedFormIds = usedForms[pid] ? usedForms[pid].map(f => f.id) : []
-    const c = await checkStartService.initialisePupilCheck(pid, checkWindow, allForms, usedFormIds, pinEnv)
+    const c = await checkStartService.initialisePupilCheck(pid, checkWindow, allForms, usedFormIds, pinEnv === 'live')
+    delete c.isLiveCheck // this not used in the check table (deprecated); but by the checkFormAllocation table
     checks.push(c)
   }
   await checkDataService.sqlCreateBatch(checks)
 }
 
+// PrepareCheck 2: Write an entry to the checkFormAllocationTable, and place a message on the queue
+// for writing to the data store used by pupil authentication.
+checkStartService.prepareCheck2 = async function (pupilIds, dfeNumber, schoolId, isLiveCheck) {
```

Comment:
The new function has been introduced but the old function is still present. Should we not be removing the old function? Should the new function not be replacing it wholesale? I am not keen on the appending of "2" - is there a need to keep the old function around... there is an implication for future work to clean up the references to the "2" function and rename this to drop the "2"?

The new function duplicates a lot of the code in the original function, which I find concerning. I feel there should be a better way to achieve this.

The new function uses the Azure queue, but I would have thought that removal of unused code and/or migration to the new version would be included, without this, maintaining which function is actually used is quite difficult.

Reply:
again, see above